### PR TITLE
Hotfix/5.47.1.fixes

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL:  https://app.asana.com/0/inbox/1157893581871899/1119619712088571/1167491198050484

**Description**:
Renaming `NotificationScheduler` introduced a bug that made it impossible for `DaggerWorkerFactory`  to create Workers. The classpath had changed and therefore no longer possible to find.

This PR fixes this issue, also adding a layer of security in `DaggerWorkerFactory` preventing it from crashing if the worker can't be found.

As part of this PR we also fixed a new issue that was introduced with the Change App Icon feature. On some devices the `themeResourceId` was not found after changing the icon, so we added a default option for that too based on the `AppTheme`

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/notification/AndroidNotificationScheduler.kt [hunk 1]
```diff
@@ -66,7 +63,7 @@ class AndroidNotificationScheduler(
     }
 
     private suspend fun scheduleInactiveUserNotifications() {
-        workManager.cancelAllWorkByTag(UNUSED_APP_WORK_REQUEST_TAG)
+        workManager.cancelUniqueWork(UNUSED_APP_WORK_REQUEST_TAG)
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/notification/AndroidNotificationScheduler.kt]: is this change deliberate?

File: app/src/main/java/com/duckduckgo/app/di/DaggerWorkerFactory.kt [hunk 2]
```diff
@@ -52,22 +56,28 @@ class DaggerWorkerFactory(
...
+                is PrivacyNotificationWorker -> injectPrivacyNotificationWorker(instance)
+                is SearchPromptNotificationWorker -> injectSearchPromptNotificationWorker(instance)
+                is StickySearchNotificationWorker -> injectStickySearchNotificationWorker(instance)
+                is DismissSearchNotificationWorker -> injectDismissSearchNotificationWorker(instance)
+                else -> Timber.i("No injection required for worker $workerClassName")
+            }
+
+            return instance
+        } catch (exception: Exception){
+            Timber.i("Worker $workerClassName could not be created")
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/di/DaggerWorkerFactory.kt]: Could switch to a higher log level. maybe `e` log level. and include the exception too?

`Timber.e(exception, "....")`

**Comment by cmonfortep** [comment]
Taking this one @CDRussell @malmstein.

File: app/src/main/java/com/duckduckgo/app/notification/AndroidNotificationScheduler.kt (hunk 1 above)
Comment (malmstein) [reply app/src/main/java/com/duckduckgo/app/notification/AndroidNotificationScheduler.kt]: nope, reverted
